src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -1142,8 +1142,9 @@ public void watchFiles(File buildFile, File outputDirectory, File testOutputDire
                             registerAll(resourceDir.getCanonicalFile().toPath(), watcher);
                             resourceMap.put(resourceDir, true);
                         }
-                    } else if (resourceMap.get(resourceDir) && !resourceDir.exists()) {
-                        warn("The resource directory " + resourceDir + "was deleted.  Restart liberty:dev mode for it to take effect.");
+                    } else {
```

Comment:
It's missing a condition.
Do one of the following instead.  The first one looks less confusing.

```
if (!resourceMap.get(resourceDir) && resourceDir.exists()) {
    // added
    ...
} else if (resourceMap.get(resourceDir) && !resourceDir.exists()) {
   // deleted
```
or
```
if (!resourceMap.get(resourceDir)) {
    if (resourceDir.exists()) {
        //added
        ...
    }
} else if (!resourceDir.exists()) {
    // deleted
```